File: src/main/java/com/mediant/api/dataexchange/model/ClientRecord.java
package com.mediant.api.dataexchange.model;

public class ClientRecord {
    private final long clientId;
    private final String name;
    private final Long oauth2CredentialsId; // nullable

    public ClientRecord(long clientId, String name, Long oauth2CredentialsId) {
        this.clientId = clientId;
        this.name = name;
        this.oauth2CredentialsId = oauth2CredentialsId;
    }

    public long getClientId() { return clientId; }
    public String getName() { return name; }
    public Long getOauth2CredentialsId() { return oauth2CredentialsId; }
}

File: python/src/pyarchiver/models/client_record.py
from dataclasses import dataclass


@dataclass
class ClientRecord:
    client_id: str
    name: str | None = None
    state: str | None = None

    def to_dict(self):
        return {'client_id': self.client_id, 'name': self.name, 'state': self.state}
